src/fu-util.c

```diff
@@ -1758,32 +1749,74 @@ fu_util_remote_modify (FuUtilPrivate *priv, gchar **values, GError **error)
 static gboolean
 fu_util_remote_enable (FuUtilPrivate *priv, gchar **values, GError **error)
 {
+	g_autoptr(FwupdRemote) remote = NULL;
 	if (g_strv_length (values) != 1) {
 		g_set_error_literal (error,
 				     FWUPD_ERROR,
 				     FWUPD_ERROR_INVALID_ARGS,
 				     "Invalid arguments");
 		return FALSE;
 	}
-	if (!fu_util_modify_remote (priv, values[0], "Enabled", "true", error))
+	remote = fwupd_client_get_remote_by_id (priv->client, values[0],
+						priv->cancellable, error);
+	if (remote == NULL)
+		return FALSE;
+	if (!fu_util_modify_remote_warning (priv, remote, error))
+		return FALSE;
+	if (!fwupd_client_modify_remote (priv->client,
+					 fwupd_remote_get_id (remote),
+					 "Enabled", "true",
+					 priv->cancellable, error))
+		return FALSE;
+
+	/* ask for permission to refresh */
+	if (fwupd_remote_get_kind (remote) != FWUPD_REMOTE_KIND_DOWNLOAD) {
+		/* TRANSLATORS: success message */
+		g_print ("%s\n", _("Successfully enabled remote"));
+		return TRUE;
+	}
+	if (!priv->assume_yes) {
```

Comment:
I think this needs `priv->no_metadata_check` or `priv->no_remote_check` to check also.  Otherwise now if you call this command from a non-interactive context it will hang IIRC.

Reply:
Agree, added.